### PR TITLE
Feature: Add support for multiple Protobuf files

### PR DESCRIPTION
**Pull request by joschi**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)

Kafka UI currently supports loading only a single Protobuf file containing all message definitions: https://github.com/provectus/kafka-ui/blob/v0.4.0/documentation/guides/Protobuf.md

While this makes the implementation to load the Protobuf files and create the schema for Kafka UI shorter, it also limits the usefulness of this feature in larger organiziations which did not migrate to a proper schema registry yet.

This change introduces the `kafka.clusters.protobufFiles` setting which takes a list of Protobuf files instead of a single file in `kafka.clusters.protobufFile`.

When both configuration settings are provided, the single file in `protobufFile` is added to the list in `protobufFiles` before loading, so that the change is backward-compatible.

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [ ] Manually (please, describe, if necessary)
- [x] Unit checks
- [ ] Integration checks
- [x] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

cat-please

### CODE REVIEW

**Review by iliax — APPROVED** [review]
@joschi thank you for very useful contribution. I left minor comments, please take a look. Also it would be nice if we support references between files (`import` in protobuf terms), but if you don't have capacity for this we will implement it separately

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/serdes/builtin/ProtobufFileSerde.java [hunk 1]
```diff
@@ -46,64 +50,116 @@ public static String name() {
...
+
+  private static Map.Entry<Descriptor, Path> getDescriptorAndPath(Map<Path, ProtobufSchema> protobufSchemas,
+                                                                  String msgName) {
+    return protobufSchemas.entrySet().stream()
+            .filter(schema -> schema.getValue() != null)
+            .filter(schema -> schema.getValue().toDescriptor(msgName) != null)
+            .map(schema -> Map.entry(schema.getValue().toDescriptor(msgName), schema.getKey()))
+            .filter(descriptor -> descriptor.getKey() != null)
+            .findFirst()
+            .orElseThrow(() -> new NullPointerException(
```
[the remark this answers is not in the document]
Comment (joschi) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/serdes/builtin/ProtobufFileSerde.java:149]: @iliax Unfortunately `Map#entry(K, V)` will throw an exception if the provided key (`K`) is `null`, so we'll have to check this before creating the `Map.Entry` instance.

We can remove the `filter(descriptor -> descriptor.getKey() != null)` step, though.

[the remark this answers is not in the document]
Comment (iliax) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/serdes/builtin/ProtobufFileSerde.java:149]: ok, I see, lets rm unnecessary checks though. (`schema.getValue() != null` also looks redundant)

**Comment by joschi** [comment]
> Also it would be nice if we support references between files (`import` in protobuf terms), but if you don't have capacity for this we will implement it separately

I would prefer doing this in a separate PR.

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/serdes/builtin/ProtobufFileSerde.java [hunk 2]
```diff
@@ -46,64 +50,118 @@ public static String name() {
   @Override
   public boolean initOnStartup(PropertyResolver kafkaClusterProperties,
                                PropertyResolver globalProperties) {
-    return kafkaClusterProperties.getProperty("protobufFile", String.class)
-        .isPresent();
+    Optional<String> protobufFile = kafkaClusterProperties.getProperty("protobufFile", String.class);
```
Comment (Haarolean) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/serdes/builtin/ProtobufFileSerde.java:53]: TODO drop deprecated protobuf single file property

**Review by Haarolean — APPROVED** [review]

**Comment by Haarolean** [comment]
@joschi thanks! much appreciate it

**Pull request merged**